backend/controller/admin/local_client.go

```diff
@@ -13,8 +22,46 @@ type localClient struct {
 	*AdminService
 }
 
+type diskSchemaRetriever struct{}
+
 func newLocalClient(ctx context.Context) *localClient {
 	cm := configuration.ConfigFromContext(ctx)
 	sm := configuration.SecretsFromContext(ctx)
-	return &localClient{NewAdminService(cm, sm)}
+	return &localClient{NewAdminService(cm, sm, &diskSchemaRetriever{})}
+}
+
+func (s *diskSchemaRetriever) GetActiveSchema(ctx context.Context) (*schema.Schema, error) {
+	path, ok := projectconfig.DefaultConfigPath().Get()
+	if !ok {
+		return nil, fmt.Errorf("no project config path available")
+	}
+	projConfig, err := projectconfig.Load(ctx, path)
+	if err != nil {
+		return nil, fmt.Errorf("could not load project config: %w", err)
+	}
+	modules, err := buildengine.DiscoverModules(ctx, projConfig.AbsModuleDirs())
+	if err != nil {
+		return nil, fmt.Errorf("could not discover modules: %w", err)
+	}
+
+	var pbModules []*schemapb.Module
+	for _, m := range modules {
+		deployDir := m.Config.Abs().DeployDir
+		schemaPath := filepath.Join(deployDir, m.Config.Schema)
```

Comment:
This is joining two absolute paths, you need to just use `m.Config.Schema` directly. Can you write an integration test for this?

Reply:
Actually disregard that, per the other comment.

Reply:
`m.Config.Schema` is a relative path, not absolute, and this code path is tested in `TestAdminService` but I can make it more explicit

Reply:
Sorry I saw you were using `.Abs()` and assumed you were using it for both. What I should have said is that  as you're using`m.Config.Abs()`, you should just use `.Schema` from that, which will be absolute already.